### PR TITLE
DOC: remove hidden submodules from sidebar

### PR DESCRIPTION
**Pull request by tupui**
Simplifies the sidebar to only present modules.

Mailing list discussion here: https://mail.python.org/archives/list/scipy-dev@python.org/thread/G7ZAF46XW6VFMGUGBVLTKJGDVSDQDE4T/

(Not sure about keeping a space for `LowLevelCallable` in the sidebar. LMK)

### CODE REVIEW

File: doc/source/reference/main_namespace.rst
Comment (tupui) [inline doc/source/reference/main_namespace.rst:36]: cc @ev-br 😃

**Comment by stefanv** [comment]
Makes more sense just to list the submodules, and not LLC, I think?

**Comment by tupui** [comment]
CircleCI seems to be super slow to upload files on AWS. They say on their status page no issues but there is clearly something happening.

**Comment by rgommers** [comment]
> Makes more sense just to list the submodules, and not LLC, I think?

The content in http://scipy.github.io/devdocs/reference/ccallback.html is useful, so if you want to unlink the doc then that content has to be moved. Fine with me to do now or to leave it alone. I'd probably go for the latter; bigger fish to fry:)

**Comment by tupui** [comment]
What I could propose otherwise is to make a hidden toctree in `doc/source/reference/main_namespace.rst` and instead of having a ref to the class in the top description, I make a ref to the module.

**Comment by rgommers** [comment]
> What I could propose otherwise is to make a hidden toctree in `doc/source/reference/main_namespace.rst` and instead of having a ref to the class in the top description, I make a ref to the module.

That breaks the regular pattern, any regular function/object should be in a table and linked directly. You could link both though, `LowLevelCallable` in the table and the other doc from regular text ("for more details on callback functions, see ...")

**Comment by tupui** [comment]
Ok then I guess we can leave this for another time.

**Comment by tupui** [comment]
Should we do anything else here? I was not clear at first with my previous comment but I don't get the suggestion and if you want me to do it or not.

**Comment by tupui** [comment]
ok CI is back green. I was starting to think somehow the change had an impact on the build speed, but no just CircleCI/AWS issue.

**Comment by tupui** [comment]
@rgommers do you want me do to anything else here?

**Comment by stefanv** [comment]
@tupui You now have module names in the sidebar, except for the first and last items (SciPy main namespace, and LowLevelCallable).  I'd rename `SciPy main namespace` to `scipy` for consistency, and add the LowLevelCallable docs to the `scipy` page to lessen confusion.
In the `scipy` member list, for LowLevelCallable you could then link to the LowLevelCallable section as well.

**Comment by stefanv** [comment]
BTW it looks much tidier :)

**Comment by mdhaber** [comment]
I was also going to suggest moving LowLevelCallable into the main SciPy namespace page. It is, after all, in the main SciPy namespace. And I'll second the motion to rename "The main SciPy namespace" to `scipy`. After that, let's merge it.

**Comment by tupui** [comment]
I did the changes. LMK 😃

**Comment by stefanv** [comment]
@tupui Should the ToC also only have the first sub-namespace, or did you want that one to go deeper, as it currently does?

Also, the content from the LowLevelCallable doc is now gone; did you mean to incorporate it elsewhere?

**Comment by tupui** [comment]
> @tupui Should the ToC also only have the first sub-namespace, or did you want that one to go deeper, as it currently does?

You mean the 2 levels on the API page? I think this is ok yes.

> Also, the content from the LowLevelCallable doc is now gone; did you mean to incorporate it elsewhere?

`LowLevelCallable` can be seen on the `scipy` page here https://output.circle-artifacts.com/output/job/56342caa-63f0-40c6-bd74-3574a0298ab0/artifacts/0/html/reference/main_namespace.html

If you click on the link which says `and one class (LowLevelCallable)`, it goes to the page of the "module". I think it's a good compromise as it's not a module.

**Comment by stefanv** [comment]
Right, but there was text on LowLevelCallable that's now missing, no?

**Comment by tupui** [comment]
For people following, we had an offline chat: I will move the content of `ccallback.rst` into the docstring of `LowLevelCallable`, this way we can remove the extra file which is not really necessary just for this content.

**Comment by tupui** [comment]
Here is the change. (The see also section is there on purpose to render properly with the correct markup.)

File: doc/API.rst.txt
Comment (mdhaber) [inline doc/API.rst.txt:181]: Looks like everything is here except what was intended to be removed (sub-subpackages, ccallback).

File: doc/source/reference/ccallback.rst
Comment (mdhaber) [inline doc/source/reference/ccallback.rst:2]: I think moving all this into the `LowLevelCallable` documentation turned out fine.

File: doc/source/reference/cluster.rst
Comment (mdhaber) [inline doc/source/reference/cluster.rst:10]: It's arguable that we want these in the main list because `cluster` has no other attributes. I'll let @stefanv confirm because he asked about this before, but I'm OK with either.

File: doc/source/reference/optimize.rst
Comment (mdhaber) [inline doc/source/reference/optimize.rst:10]: This is now available in the `optimize` page as:

image

and

image

It's not super prominent, but maybe that's OK. Should it be listed along with `show_options`, `OptimizeResult`, and `OptimizeWarning` at the top? It looks like the choice to make the `toctree` hidden was intentional. Fine with me either way.

File: doc/source/reference/spatial.rst
Comment (mdhaber) [inline doc/source/reference/spatial.rst:9]: This seems fine. `spatial.distance` and `spatial.transform` are clearly linked from `scipy.spatial`.

File: scipy/linalg/__init__.py
Comment (mdhaber) [inline scipy/linalg/__init__.py:15]: As with `scipy.optimize.cython_optimize`, these are now only mentioned in "See Also" sections. Maybe they should appear in a top-level listing on the `linalg` page? It looks like the choice to make the `toctree` hidden was intentional. Fine with me either way.

**Review by mdhaber — APPROVED** [review]
I wouldn't ask for any changes, but I'll wait for another approval to merge since others had comments.

**Comment by tupui** [comment]
@rgommers @stefanv is this good for you?

**Comment by stefanv** [comment]
This is a good set of improvements; we can always adjust nitpicks on a follow-up PR. Thanks, @tupui!

**Pull request merged**
